evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -36,16 +40,21 @@ def _make_entity_set(self, X):
         ft_es = EntitySet()
         # TODO: This delete was introduced for compatibility with Featuretools 1.0.0.  This should
         # be removed after Featuretools handles unnamed dataframes being passed to this function.
+        # But even then, we should still use any dataframe name that might be available from input features.
         del X.ww
         if self.index not in X.columns:
             es = ft_es.add_dataframe(
                 dataframe=X,
-                dataframe_name="X",
+                dataframe_name=self._dataframe_name,
                 index=self.index,
                 make_index=True,
             )
         else:
-            es = ft_es.add_dataframe(dataframe=X, dataframe_name="X", index=self.index)
+            es = ft_es.add_dataframe(
+                dataframe=X,
+                dataframe_name=self._dataframe_name,
+                index=self.index,
+            )
```

Comment:
This is a bit of an unrelated nitpick, but I think this entire code block can be cleaned up to remove the if/else construct:

```
del X.ww
should_make_index = self.index not in X.columns
es = ft_es.add_dataframe(
    dataframe=X,
    dataframe_name=self._dataframe_name,
    index=self.index,
    make_index=should_make_index,
)
```

Reply:
nice, changed